src/hooks/useFetchAssetImages.ts

```diff
@@ -1,22 +1,36 @@
 import { useState, useEffect } from "react";
 import firebase from "firebase/app";
 
-export const useFetchAssetImages = (path: string): string[] => {
-  const [imagePaths, setImagePaths] = useState<string[]>([]);
+export interface UseFetchAssetsReturn {
+  assets: string[];
+  isLoading: boolean;
+}
+
+/**
+ *
+ * @param path: Additional path parameter that is used in addition to the `assets/` path in the database storage.
+ * @example useFetchAssets("icons")
+ * @returns All assets from the provided firebase storage path.
+ */
+export const useFetchAssets = (path: string): UseFetchAssetsReturn => {
+  const [assets, setAssets] = useState<string[]>([]);
+  const [isLoading, setLoading] = useState<boolean>(false);
 
   useEffect(() => {
-    const fetchImages = async () => {
+    const fetchAssets = async () => {
+      setLoading(true);
       const storageRef = firebase.storage().ref();
 
       const list = await storageRef.child(`assets/${path}`).listAll();
       const promises = list.items.map((item) => item.getDownloadURL());
 
       const urls: string[] = await Promise.all(promises);
 
-      setImagePaths(urls);
+      setAssets(urls);
     };
-    fetchImages();
+
+    fetchAssets().finally(() => setLoading(false));
   }, [path]);
 
-  return imagePaths;
+  return { assets, isLoading };
 };
```

Comment:
useAsyncFn

Reply:
Fixed in: https://github.com/sparkletown/sparkle/pull/1583/commits/f8fa0c67c34ac626919737b4bbd112b4bd715553